### PR TITLE
TRCL-3691 v4-client: Add signing function for push 

### PR DESCRIPTION
**Pull request by ruixhuang**
Add native function to sign push notification token registration payload.  Also add function to derive keys from mnemonic.  This also fixes the function signCompliancePayload().

### CODE REVIEW

File: v4-client-js/src/clients/native.ts [hunk 1]
```diff
@@ -1306,6 +1312,34 @@ export async function signCompliancePayload(payload: string): Promise<string> {
   }
 }
 
+export async function signPushNotificationTokenRegistrationPayload(payload: string): Promise<string> {
```
Comment (prashanDYDX) [inline v4-client-js/src/clients/native.ts:1315]: should we just have a generic sign payload function? is there specific handling needed for either the push case or compliance case?

File: v4-client-js/src/clients/native.ts [hunk 2]
```diff
@@ -147,6 +147,12 @@ export async function connectWallet(mnemonic: string): Promise<string> {
     globalThis.wallet = await LocalWallet.fromMnemonic(mnemonic, BECH32_PREFIX);
     globalThis.nobleWallet = await LocalWallet.fromMnemonic(mnemonic, NOBLE_BECH32_PREFIX);
 
+    const { privateKey, publicKey } = deriveHDKeyFromMnemonic(mnemonic);
```
Comment (rosepuppy) [inline v4-client-js/src/clients/native.ts:150]: we are already setting the hdKey in `deriveMnemomicFromEthereumSignature`, are we able to use it?

This way we shouldn't need to make `deriveHDKeyFromMnemonic`

**Review by prashanDYDX — APPROVED** [review]

File: v4-client-js/src/clients/native.ts (hunk 2 above)
Comment (rosepuppy) [reply v4-client-js/src/clients/native.ts:150]: ohhh ok TIL

File: v4-client-js/src/lib/onboarding.ts [hunk 3]
```diff
@@ -67,3 +67,38 @@ export const deriveHDKeyFromEthereumSignature = (
...
+ *
+ * @param mnemonic used to generate seed
+ *
+ * @param path BIP44 HD Path. Default is The Cosmos Hub path
+ *
+ * @throws Error if the hdkey does not exist
+ *
+ * @returns Priv/pub keys
+ */
+export const deriveHDKeyFromMnemonic = (
```
Comment (rosepuppy) [inline v4-client-js/src/lib/onboarding.ts:84]: nit: maybe we can call this from `exportMnemonicAndPrivateKey`

**Review by rosepuppy — APPROVED** [review]

**Pull request merged**